sdk/tracing/src/test/java/io/opentelemetry/sdk/trace/SamplersTest.java

```diff
@@ -231,13 +261,75 @@ void parentOrElseSampler_AlwaysOn() {
                     Collections.emptyList())
                 .getDecision())
         .isEqualTo(Decision.RECORD_AND_SAMPLED);
+
+    assertThat(
+            Samplers.parentBased(Samplers.alwaysOn(), null, null, null, Samplers.alwaysOn())
+                .shouldSample(
+                    invalidSpanContext,
+                    traceId,
+                    SPAN_NAME,
+                    SPAN_KIND,
+                    Attributes.empty(),
+                    Collections.emptyList())
+                .getDecision())
+        .isEqualTo(Decision.RECORD_AND_SAMPLED);
+
+    // Sampled remote parent
```

Comment:
What do you think about breaking this up into smaller tests, so they will fail/pass individually, rather than having one giant test?

Reply:
sure :)